packages/libs/react-ui/src/components/Accordion/AccordionSection.tsx

```diff
@@ -3,55 +3,61 @@
 import {
   accordionContentWrapperClass,
   accordionSectionClass,
+  accordionSectionHeadingClass,
   accordionTitleClass,
   accordionTitleVariants,
   toggleButtonClass,
 } from './Accordion.css';
 
 import { SystemIcon } from '@components/Icon';
 import classNames from 'classnames';
-import React, { FC, useEffect, useRef } from 'react';
+import type { FC } from 'react';
+import React from 'react';
 
 export interface IAccordionSectionProps {
-  title: React.ReactNode;
-  children: React.ReactNode;
-  isOpen: boolean;
-  onToggle: () => void;
-  onOpen?: () => void;
+  children?: React.ReactNode;
+  index?: number;
+  isOpen?: boolean;
+  onClick?: () => void;
   onClose?: () => void;
+  onOpen?: () => void;
+  title: string;
 }
 
 export const AccordionSection: FC<IAccordionSectionProps> = ({
-  isOpen = false,
-  title,
   children,
-  onToggle,
-  onOpen,
+  isOpen,
+  onClick,
   onClose,
+  onOpen,
+  title,
 }) => {
-  const didMountRef = useRef(false);
-
-  useEffect(() => {
-    if (!didMountRef.current) {
-      didMountRef.current = true;
-      return;
+  const handleClick = (): void => {
+    if (isOpen) {
+      onClose?.();
+    } else {
+      onOpen?.();
     }
-
-    if (isOpen) onOpen?.();
-    else onClose?.();
-  }, [isOpen]);
-
+    onClick?.();
+  };
   return (
-    <div className={accordionSectionClass}>
+    <section
+      className={accordionSectionClass}
+      data-testid="kda-accordion-section"
+    >
       <div
```

Comment:
should this be a button? and then you can wrap the icon in a span

Reply:
The plus is already a button so then it would be a button in a button. But I'm not sure what would be semantically ideal..

Reply:
Well right now the icon is a button, the onClick handler is placed on the div wrapping the text so this can be semantically improved. Spoke with Isa and the whole text + icon should trigger the interaction so I think the html needs to be reworked a bit

Reply:
Oh, currently the whole text + icon already triggers the interaction so that's not an issue but I agree we can improve the html structure. I'll do that :)

Reply:
How about we turn the sections into [articles](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/article) and have the title and icon wrapped in a <header>? I'll push that change so you can see.